flang/lib/Optimizer/Analysis/AliasAnalysis.cpp

```diff
@@ -372,6 +372,17 @@ getAttrsFromVariable(fir::FortranVariableOpInterface var) {
   return attrs;
 }
 
+static std::optional<omp::BlockArgOpenMPOpInterface>
+getOpenMPBlockArgInterface(Operation *op) {
+  std::optional<omp::BlockArgOpenMPOpInterface> blockArgOpenMPOpInterface;
+  if (!op)
+    return blockArgOpenMPOpInterface;
+  if (llvm::isa<omp::TargetOp>(op) || llvm::isa<omp::ParallelOp>(op)) {
```

Comment:
Why not return the block args interface for other operation types (where the dyn_cast succeeds)?